src/platforms/hosted/cmsis_dap.c

```diff
@@ -454,34 +454,37 @@ static void dap_mem_write_sized(adiv5_access_port_s *ap, uint32_t dest, const vo
 {
 	if (len == 0)
 		return;
-	DEBUG_WIRE("memwrite @ %" PRIx32 " len %zu, align %d, data: %08x \n", dest, len, align, *(uint32_t *)src);
-	if (((unsigned)(1 << align)) == len)
+	DEBUG_WIRE("memwrite @ %" PRIx32 " len %zu, align %d, data: %08x \n", dest, len, align, *(const uint32_t *)src);
+	/* If the write can be done in a single transaction, use the dap_write_single() fast-path */
+	if ((1U << align) == len)
 		return dap_write_single(ap, dest, src, align);
-	unsigned int max_size = ((report_size - 6U) >> (2U - align) & ~3U);
-	while (len) {
-		dap_ap_mem_access_setup(ap, dest, align);
-		unsigned int blocksize = (dest | 0x3ffU) - dest + 1U;
-		if (blocksize > len)
-			blocksize = len;
-		while (blocksize) {
-			unsigned int transfersize = blocksize;
-			if (transfersize > max_size)
-				transfersize = max_size;
-			if (!dap_write_block(ap, dest, src, transfersize, align)) {
-				DEBUG_WARN("mem_write failed %u\n", ap->dp->fault);
-				ap->dp->fault = 1;
+	/* Otherwise proceed blockwise */
+	dap_ap_mem_access_setup(ap, dest, align);
```

Comment:
This needs to be called inside the loop with `dest + offset` to update AP_TAR at 1024B boundaries, otherwise it'll wrap.

Reply:
This should now be fixed - many thanks :)